src/data/write_init_files.py

```diff
@@ -949,8 +959,27 @@ def write_phys_read_subroutine(outfile, host_dict, host_vars, host_imports,
     outfile.blank_line()
     outfile.comment("If an index was found in the constituent hash table, then read in the data to that index of the constituent array", 6)
     outfile.blank_line()
+    outfile.write("var_found = .false.", 6)
     outfile.write("field_data_ptr => cam_constituents_array()", 6)
-    outfile.write("call read_field(file, ccpp_required_data(req_idx), [ccpp_required_data(req_idx)], 'lev', timestep, field_data_ptr(:,:,constituent_idx), mark_as_read=.false.)", 6)
+    outfile.write("call read_field(file, ccpp_required_data(req_idx), [ccpp_required_data(req_idx)], 'lev', timestep, field_data_ptr(:,:,constituent_idx), mark_as_read=.false., error_on_not_found=.false., var_found=var_found)", 6)
+    outfile.write("if(.not. var_found) then", 6)
+    outfile.write("const_props => cam_model_const_properties()", 7)
+    outfile.write("constituent_has_default = .false.", 7)
+    outfile.write("call const_props(constituent_idx)%has_default(constituent_has_default, constituent_errflg, constituent_errmsg)", 7)
+    outfile.write("if (constituent_has_default) then", 7)
```

Comment:
That's a good idea.  Is there any preferred behavior if it's set?  Error and exit, log and set to 0, etc.?

Reply:
Yep, for now I think anytime `constituent_errflg` is non-zero we should kill the model.  This can be done by calling `endrun` and passing in the `constituent_errmsg` as the `endrun` message input argument.  There should be `endrun` examples all over CAM-SIMA, but if you want a specific example just let me know.

Reply:
Updated but let me know if there's more needed in the error message output.